src/interfaces.ts

```diff
@@ -106,93 +130,93 @@ interface RouteConfig {
+  canActivate(params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction): boolean | Promise<boolean> | PromiseLike<boolean> | NavigationCommand | Promise<NavigationCommand> | PromiseLike<NavigationCommand>;
 }
 
 /**
 * An optional interface describing the activate convention.
 */
-interface RoutableComponentActivate {
+export interface RoutableComponentActivate {
   /**
   * Implement this hook if you want to perform custom logic just before your view-model is displayed.
   * You can optionally return a promise to tell the router to wait to bind and attach the view until
   * after you finish your work.
   */
-  activate: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => Promise<void>|PromiseLike<void>|IObservable|void;
+  activate(params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction): Promise<void> | PromiseLike<void> | IObservable | void;
 }
 
 /**
 * An optional interface describing the canDeactivate convention.
 */
-interface RoutableComponentCanDeactivate {
+export interface RoutableComponentCanDeactivate {
   /**
   * Implement this hook if you want to control whether or not the router can navigate away from your
   * view-model when moving to a new route. Return a boolean value, a promise for a boolean value,
   * or a navigation command.
   */
-  canDeactivate: () => boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+  canDeactivate: () => boolean | Promise<boolean> | PromiseLike<boolean> | NavigationCommand;
 }
 
 /**
 * An optional interface describing the deactivate convention.
 */
-interface RoutableComponentDeactivate {
+export interface RoutableComponentDeactivate {
   /**
   * Implement this hook if you want to perform custom logic when your view-model is being
   * navigated away from. You can optionally return a promise to tell the router to wait until
   * after you finish your work.
   */
-  deactivate: () => Promise<void>|PromiseLike<void>|IObservable|void;
+  deactivate: () => Promise<void> | PromiseLike<void> | IObservable | void;
 }
 
 /**
 * An optional interface describing the determineActivationStrategy convention.
 */
-interface RoutableComponentDetermineActivationStrategy {
+export interface RoutableComponentDetermineActivationStrategy {
   /**
   * Implement this hook if you want to give hints to the router about the activation strategy, when reusing
   * a view model for different routes. Available values are 'replace' and 'invoke-lifecycle'.
   */
-  determineActivationStrategy: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => 'no-change' | 'invoke-lifecycle'| 'replace';
+  determineActivationStrategy(params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction): 'no-change' | 'invoke-lifecycle' | 'replace';
```

Comment:
Should the return type also use a `keyof`?

Reply:
yes, I missed this. Should be `ActivationStrategy[keyof ActivationStrategy]`